### ISSUE TITLE
Implement long description link to table

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#858]
**Page 32 in DAC report**

**Issue from DAC report:**
The table provides a suitable alternative to the information displayed within the chart. However, the table is not presented to users as a direct alternative, as there is no clear label to allow users to identify its purpose as an alternative. Users may not understand the association between the alternative forms of information.

**Solution from DAC report:**
Consider implementing a long description link to navigating users to the tabular form of the data using the ‘longdesc’ attribute. Please consider the implementation below in order to ensure that the context of the actionable components associated with the chart are contextualised as part of the chart group.

**Example from DAC report:**

```
<figure role="group">
    <figcaption class="visuallyhidden"> Percentage of the
 population living in households at risk of poverty </figcaption>
    <div id="chart-canvas">
        <h3 id="chart-heading" class="chart-title">Percentage of the
 population living in households at risk of poverty</h3>
        <div id="chart" class="plot-container" style="height: 420px;">
            <div class="chartjs-size-monitor">
                <div class="chartjs-size-monitor-expand"><div class=""></div></div>
                <div class="chartjs-size-monitor-shrink"><div class=""></div></div>
            </div>
            <canvas role="img" aria-labelledby="chart-heading plotLegend"
 style="display: block; height: 400px; width: 795px;" width="1192"
 height="600" class="chartjs-render-monitor" longdesc="link to table"
 ></canvas>
    …
    </div>
    <div id="plotLegend">
        <ul id="legend">
            <li data-datasetindex="0">
                <button aria-label="Remove graph of United Kingodom from
 chart" class="swatch" style="background-color: #777777">
 United Kingdom</button>
            </li>
        </ul>
    </div>
    <div id="selectionChartFooter" class="table-footer-text">
            <dl>
                <dt>Source: </dt><dd>Eurostat</dd>
                <dt>Geographical Area: </dt><dd>United Kingdom</dd>
                <dt>Unit of Measurement: </dt><dd>Percentage (%)</dd>
            </dl>
        </div>
    </div>
</figure>
```

Additionally, consider the following implementation so that screen reader users know where to access the alternative (for chart and map tab)

```
<div style="display: none;">
    <span id="details">For tabular data alternative see Table tab</span>
</div>
<ul class="nav nav-tabs data-view" role="tablist">
    <li role="presentation" class="nav-item active">
        <a class="nav-link" data-toggle="tab" href="#chartview"
 aria-controls="chartview" role="tab" data-on="click" data-event-
 category="Tab change" data-event-action="Change data view" data-event-
 label="Change to Chart tab" aria-describedby="details">Chart</a>
    </li>
    …
</ul>
```

**Comment by brockfanning** [comment]
I'm noting this question to ask DAC at some point: I'm confused on what "link to table" should be. The table is on the same page and it is a dynamic table that responds to sub-categories selections (just as the chart does). In other words, I feel like ideally we need some way for screen-reader users to activate the table tab (the equivalent of clicking "Table"). Is this what "longdesc" will accomplish? Or will the "longdesc" prompt the screen-reader to narrate the contents of the table, at that moment.

**Comment by LucyGwilliamAdmin** [comment]
Just noting DAC's reply to Brock's query here:

>The longdesc attribute is a URI, please see [H45: Using longdesc](https://www.w3.org/WAI/WCAG21/Techniques/html/H45). >This can be a same page link or an external link. The logistics of using this attribute to actually trigger the Table tab to open, I am not sure about. I think the easiest implementation would be to have a separate page which just contains the same contents as the Table tab (from #selectionsTable) which is then linked to in the longdesc attribute. Screen reader users would then have the option to view the longdesc as they encounter the image. Then the actual Table tab is also available as it is at the moment, if users wish to navigate directly to the Table that way.

>Here are some additional resources:

>[G73: Providing a long description in another location with a link to it that is immediately adjacent to the non-text content](https://www.w3.org/WAI/WCAG21/Techniques/general/G73)
>[G74: Providing a long description in text near the non-text content, with a reference to the location of the long description in the short description](https://www.w3.org/WAI/WCAG21/Techniques/general/G74)
>[G92: Providing long description for non-text content that serves the same purpose and presents the same information](https://www.w3.org/WAI/WCAG21/Techniques/general/G92)

@AnnCorp @brockfanning @phillipgardiner @jwestw I think this may be one to discuss at tomorrow's tech call, as looks like DAC are suggesting adding a separate page which contains the same info that is on the table tab, which to me seems a little strange.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @AnnCorp 
In #894 we added `aria-describedby` for this purpose, instead of `longdesc`. We tried `longdesc` but the screenreaders didn't seem to even notice it. I think we can close this unless DAC feels strongly that `longdesc` should be added anyway. (Which would be an easy change.)

**Comment by LucyGwilliamAdmin** [comment]
Feedback from DAC:

> This is looking absolutely fab to me! I think it works really well!

So closing this one 👍
